sockeye/inference.py

```diff
@@ -1009,29 +1013,25 @@ def _beam_search(self,
                                                                        models_output_layer_w,
                                                                        models_output_layer_b)
 
-            # (2) compute length-normalized accumulated scores in place
-            if t == 1 and self.batch_size == 1:  # only one hypothesis at t==1
-                scores = scores[:1] / self.length_penalty(lengths[:1])
-            else:
-                # renormalize scores by length ...
-                scores = (scores + scores_accumulated * self.length_penalty(lengths - 1)) / self.length_penalty(lengths)
```

Comment:
why does this change not affect the search behavior? If you don't normalize hypotheses by length, the results of smallest_k will be different, or am I missing something?

Reply:
It is the same until you have mixed finished / unfinished hypotheses. At that point, the finished ones have a much lower score and jump to the top. If you use pruning, they make quick work of the beam, throwing out most hypotheses.